src/main/java/com/ibm/watson/modelmesh/ModelMesh.java

```diff
@@ -5541,147 +5488,208 @@ private Runnable rateTrackingTask() {
             // applies only to time-tracking mode, updated after each non-empty iteration
             double averageModelParallelism = 1.0;
 
+            // this is incremented each iteration (~ every RATE_CHECK_INTERVAL_MS)
+            int iterationCounter;
+
             @Override
             public void run() {
                 Thread curThread = Thread.currentThread();
                 String threadNameBefore = setThreadName(curThread, "rate-track-task");
                 try {
-                    long lastTime = lastCheckTime, now = currentTimeMillis();
-                    long timeDelta = now - lastTime; // should be ~= RATE_CHECK_INTERVAL_MS
+                    final long lastTime = lastCheckTime, now = currentTimeMillis();
+                    final long timeDelta = now - lastTime; // should be ~= RATE_CHECK_INTERVAL_MS
 
                     // skip if too soon - i think this can happen if task executions
                     // get backed up in the taskPool
                     if (timeDelta * 5L < RATE_CHECK_INTERVAL_MS * 3L) {
                         return;
                     }
 
-                    int instCount = clusterStats.instanceCount;
-                    if (instCount <= 2) {
-                        return; // can't scale beyond 2 copies regardless
-                    }
+                    // Determine the upper and lower bounds (inclusive) in terms of
+                    // iteration number between which a prior use of a single-copy model
+                    // will trigger a second copy to be loaded.
+                    final int lower = iterationCounter - SECOND_COPY_MAX_AGE_ITERS;
+                    final int upper = iterationCounter - SECOND_COPY_MIN_AGE_ITERS;
 
-                    final long before = logger.isDebugEnabled() ? nanoTime() : 0L;
+                    boolean success = false;
```

Comment:
What does `success` mean? Is this `true` both if we don't need to scale and didn't, or needed to scale and did?

Reply:
No it's more to ensure we don't progress the common `lastCheckTime` if we didn't reset all of the individual counters of the cache entries used within the last time interval. I.e. if there was an unexpected exception meaning that we aborted prematurely. But looking closer at the nesting/ordering I don't think it's really necessary so I'll just remove it, thanks asking about it!